src/components/Main/Header.js

```diff
@@ -0,0 +1,139 @@
+
+  const {
+    getCookie,
+    state: { isLoggedIn }
+  } = useCookie();
+
+  const navMenu = [
+    { path: '/', title: 'Home' },
+    { path: '/products', title: 'Shop' },
+    { path: '/contact', title: 'Contact' }
+  ];
+
+  const [isActive, setIsActive] = useState(window.location.pathname);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const search = (searchTerm) => {
+    navigate(`/products/filter?name=${searchTerm}`);
+  };
+
+  useEffect(() => {
+    getCookie();
+  }, []);
+
+  return (
+        <header className="header-section">
+            <div className="container">
+                <div className="inner-header">
+                    <div className="row">
+                        <div className="col-lg-2 col-md-2">
+                            <div className="logo">
+                                <NavLink to="/">
+                                    <img src={Logo} alt="Logo" />
+                                </NavLink>
+                            </div>
+                        </div>
+                        <div className="col-lg-7 col-md-7">
+                            <div className="advanced-search">
+                                <div className="input-group">
+                                    <input
+                                        type="text"
+                                        placeholder="What do you need?"
+                                        value={searchTerm}
+                                        onChange={(e) =>
+                                          setSearchTerm(e.target.value)
+                                        }
+                                    />
+                                    <button
+                                        type="button"
+                                        onClick={() => search(searchTerm)}
+                                    >
+                                        <i className="ti-search"></i>
+                                    </button>
+                                </div>
+                            </div>
+                        </div>
+                        <div className="col-lg-3 text-right col-md-3">
+                            <ul className="nav-right">
+                                {' '}
+                                {isLoggedIn
+                                  ? (
+                                    <>
+                                        <i className="fa fa-cart-arrow-down"></i>{' '}
+                                        Shopping Cart
+                                        <br />
+                                        <i className="fa fa-user-circle"></i>{' '}
+                                        Profile <br />
+                                        <NavLink
+                                            to="/logout"
+                                            onClick={() =>
+                                              setIsActive('/logout')
```

Comment:
instead of calling setIsActive on click, it's better to add useEffect on url change and call this method from there